unity_sds_client/services/data_service.py

```diff
@@ -66,8 +66,14 @@ def get_collection_data(self, collection: type= Collection):
 
         for dataset in results:
             ds = Dataset(dataset['id'], collection.collection_id, dataset['properties']['start_datetime'], dataset['properties']['end_datetime'], dataset['properties']['created'])
-            ds.add_data_file(DataFile("data" ,dataset['assets']['data']['href']))
-            ds.add_data_file(DataFile("metadata" ,dataset['assets']['metadata__data']['href']))
+
+            for filename in dataset['assets']:
+                location = dataset['assets'][filename]['href']
+                file_type = dataset['assets'][filename]['type']
+                title = dataset['assets'][filename]['title']
+                description = dataset['assets'][filename]['description']
```

Comment:
Type, Title, and Description are all optional elements. What will happen here if they aren't provided?

Reply:
The problem with this code block is we are parsing STAC from U-DS differently than we are parsing STAC from the to/from_stac methods. in the to/from_stac methods, we map the DataFile.type property to an asset_role, here we are mapping it to the asset.type property, which are not the same and will cause issues here. (e.g. the 'type' of a file might be "application/netcdf" but if we call  "to_stac" on that, we'll get some bizarre asset-roles.

here we should either treat this like the CMR query (where we look for asset roles and if they are there use them, if not, check the "key" if it looks like "data" or "metadata" then make that the filetype, but we cannot mix/match these.